packages/expo-image-picker/src/ExponentImagePicker.web.ts

```diff
@@ -92,44 +91,23 @@ function openFileBrowserAsync({
   document.body.appendChild(input);
 
   return new Promise((resolve, reject) => {
-    input.addEventListener('change', () => {
+    input.addEventListener('change', async () => {
       if (input.files) {
-        const targetFile = input.files[0];
-        const reader = new FileReader();
-        reader.onerror = () => {
-          reject(new Error(`Failed to read the selected media because the operation failed.`));
-        };
-        reader.onload = ({ target }) => {
-          const uri = (target as any).result;
-
-          const returnRaw = () => {
-            resolve({
-              cancelled: false,
-              uri,
-              width: 0,
-              height: 0,
-            });
-          };
-          if (typeof target?.result === 'string') {
-            const image = new Image();
-            image.src = target.result;
-            image.onload = function() {
-              resolve({
-                cancelled: false,
-                uri,
-                width: image.naturalWidth ?? image.width,
-                height: image.naturalHeight ?? image.height,
-              });
-            };
-            image.onerror = () => {
-              returnRaw();
-            };
-          } else {
-            returnRaw();
+        if (allowsMultipleSelection === false) {
+          const result = await readFile(input.files[0])
+          resolve(result)
+        } else {
+          const list: ImagePickerResult[] = []
+          for (let i = 0; i < input.files.length; i++) {
+            list.push(await readFile(input.files[i]))
+
           }
-        };
-        // Read in the image file as a binary string.
-        reader.readAsDataURL(targetFile);
+          resolve(list)
```

Comment:
Just as @barthap suggested go with sth like this:
```suggestion
resolve(Promise.all(input.files.map(readFile)));
```